docs/core/integration-quality-scale/rules/runtime-data.md

```diff
@@ -35,6 +36,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: MyIntegrationConfigEntry
     return True
 ```
 
+:::info
+If the integration implements `strict-typing`, the use of a custom typed `MyIntegrationConfigEntry` is required and must be used throughout.
```

Comment:
I'm not following why this needs to be documented at all?

If strict typing is enabled it would already been enforced?

Reply:
> I'm not following why this needs to be documented at all?
> 
> If strict typing is enabled it would already been enforced?

No it wouldn't be
Mypy will be quite happy with `ConfigEntry` or `ConfigEntry[Any]`

If you look at the corresponding core PR you will see that it keeps coming up with violations  as I rebase it.

Reply:
> Mypy will be quite happy with ConfigEntry or ConfigEntry[Any]

But that occurs on many places? I'm not sure how that is more of a concern in this case to compared to any other case using Any in places?